### PR TITLE
Split out VHLO contribution guidelines into their own file

### PR DESCRIPTION
**Pull request by mlevesquedion**
This may help with making sure we don't miss any items when making changes to VHLO. The main vhlo.md file should focus on what VHLO is.

### CODE REVIEW

**Review by GleasonK — APPROVED** [review]

File: docs/vhlo.md [hunk 1]
```diff
@@ -1,5 +1,7 @@
 # The VHLO Dialect
 
+See `vhlo_checklist.md` for steps to follow when making changes to VHLO.
```
Comment (ghpvnist) [inline docs/vhlo.md:3]: Link?

File: docs/vhlo_checklist.md [hunk 2]
```diff
@@ -0,0 +1,72 @@
+# VHLO Checklist
+
+See `vhlo.md` for information about VHLO.
```
Comment (ghpvnist) [inline docs/vhlo_checklist.md:3]: Link?

**Review by ghpvnist — APPROVED** [review]
LGTM with comments

File: docs/vhlo.md (hunk 1 above)
Comment (mlevesquedion) [reply docs/vhlo.md]: Done, thanks!

File: docs/vhlo_checklist.md (hunk 2 above)
Comment (mlevesquedion) [reply docs/vhlo_checklist.md]: Done, thanks!

**Pull request merged**